autopush/router/apnsrouter.py

```diff
@@ -139,31 +140,39 @@ def _route(self, notification, router_data):
                 "alert": {"title": " ", "body": " "}
             })
         apns_id = str(uuid.uuid4()).lower()
-        try:
-            apns_client.send(router_token=router_token, payload=payload,
-                             apns_id=apns_id)
-        except (ConnectionError, AttributeError) as ex:
-            self.metrics.increment("notification.bridge.error",
-                                   tags=make_tags(self._base_tags,
-                                                  application=rel_channel,
-                                                  reason="connection_error"))
-            self.log.error("Connection Error sending to APNS",
-                           log_failure=Failure(ex))
+        # APNs may force close a connection on us without warning.
+        # if that happens, retry the message.
+        success = False
+        for attempt in range(0, self._max_retry):
+            try:
+                apns_client.send(router_token=router_token, payload=payload,
+                                 apns_id=apns_id)
+                success = True
+                break
+            except AttributeError as ex:
```

Comment:
We can kill the AttributeError handling in here and the associated one in apns2 at this point. It was from that old hyper bug that was eventually fixed. Doesn't have to be in this PR, though